<!--REQUEST_ID=0ef525d5-945a-4208-aac5-b5761929dd6a-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.47023&attempt_id=a26cada8-89a6-4882-9e6e-a8d8e735d50c)

Attempted: nethack, nethack-qt

The following builds were skipped because they don't evaluate on x86_64-darwin: pkgsCross.raspberryPi.nethack, pkgsCross.aarch64-multiplatform.nethack

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/bin/moc -o qt4plsel.moc ../win/Qt4/qt4plsel.h
/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/bin/moc -o qt4set.moc ../win/Qt4/qt4set.h
/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/bin/moc -o qt4stat.moc ../win/Qt4/qt4stat.h
[CXX] ../win/Qt4/qt4str.cpp
/nix/store/cblfnvb5rmhd2z231mqasn0brzh1hhv4-bash-4.4-p23/bin/bash: g++: command not found
make[1]: *** [Makefile:920: qt4str.o] Error 127
make[1]: Leaving directory '/private/tmp/nix-build-nethack-qt-3.6.1.drv-0/nethack-3.6.1/src'
make: *** [Makefile:155: nethack] Error 2
builder for '/nix/store/y393i5nznyipkr0g08hzs18gdkf5psai-nethack-qt-3.6.1.drv' failed with exit code 2
error: build of '/nix/store/y393i5nznyipkr0g08hzs18gdkf5psai-nethack-qt-3.6.1.drv' failed
```
</p></details>

--- a/pkgs/games/nethack/default.nix
+++ b/pkgs/games/nethack/default.nix
@@ -1,7 +1,7 @@
 { stdenv, lib, fetchurl, coreutils, ncurses, gzip, flex, bison
 , less, makeWrapper
 , buildPackages
-, x11Mode ? false, qtMode ? false, libXaw, libXext, mkfontdir, pkgconfig, qt5
+, x11Mode ? false, qtMode ? false, libXaw, libXext, bdftopcf, mkfontdir, pkgconfig, qt5
 }:
 
 let
@@ -34,10 +34,11 @@ in stdenv.mkDerivation rec {
                 ++ lib.optionals qtMode [ gzip qt5.qtbase.bin qt5.qtmultimedia.bin ];
 
   nativeBuildInputs = [ flex bison ]
-                      ++ lib.optionals x11Mode [ mkfontdir ]
+                      ++ lib.optionals x11Mode [ mkfontdir bdftopcf ]
                       ++ lib.optionals qtMode [
                            pkgconfig mkfontdir qt5.qtbase.dev
                            qt5.qtmultimedia.dev makeWrapper
+                           bdftopcf
                          ];
 
   makeFlags = [ "PREFIX=$(out)" ];

--- a/pkgs/games/nethack/default.nix
+++ b/pkgs/games/nethack/default.nix
@@ -132,7 +132,8 @@ in stdenv.mkDerivation rec {
     chmod +x $out/bin/nethack
     ${lib.optionalString x11Mode "mv $out/bin/nethack $out/bin/nethack-x11"}
     ${lib.optionalString qtMode "mv $out/bin/nethack $out/bin/nethack-qt"}
-    install -Dm 555 util/{makedefs,dgn_comp,lev_comp,dlb} -t $out/libexec/nethack/
+    install -Dm 555 util/{makedefs,dgn_comp,lev_comp} -t $out/libexec/nethack/
+    ${lib.optionalString (!(x11Mode || qtMode)) "install -Dm 555 util/dlb -t $out/libexec/nethack/"}
   '';
 
   postFixup = lib.optionalString qtMode ''

--- a/pkgs/games/nethack/default.nix
+++ b/pkgs/games/nethack/default.nix
@@ -73,7 +73,11 @@ in stdenv.mkDerivation rec {
         -DCOMPRESS_EXTENSION=\\".gz\\",' \
       -i sys/unix/hints/linux-qt4
     ''}
-    ${lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    ${lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform)
+    # If we're cross-compiling, replace the paths to the data generation tools
+    # with the ones from the build platform's nethack package, since we can't
+    # run the ones we've built here.
+    ''
     ${buildPackages.perl}/bin/perl -p \
       -e 's,[a-z./]+/(makedefs|dgn_comp|lev_comp|dlb)(?!\.),${buildPackages.nethack}/libexec/nethack/\1,g' \
       -i sys/unix/Makefile.*

